### PR TITLE
Improving class structure of sync_data & adding unit tests

### PR DESCRIPTION
**Pull request by raun**
Improving the structure of Command class in sync_data.py so that it can be tested. Adding unit test for the same. Limiting the scope of atomic lock to writing to database. Check if the QR directory is present, if not create it.

### CODE REVIEW

File: tests/unit/tickets/management/commands/test_sync_data.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+def test_suite_command():
+	try:
+		command_case(none_events, none_order)
+	except TypeError:
+		pass
+
+	try:
+		command_case(single_event, none_order)
+	except TypeError:
+		pass
```
Comment (theskumar) [inline tests/unit/tickets/management/commands/test_sync_data.py]: Probably you wanted to use:

```
with pytest.raises(TypeError):
  command_case(none_events, none_order)

with pytest.raises(TypeError):
  command_case(single_event, none_order)
```

**Comment by theskumar** [comment]
Hi @raun, this project usages PEP8 as code style guide, checked via flake8[1]. This is causing the tests to fail.

Please resolve the PEP8 issues/warnings, so we can merge this. If you like you can use https://pypi.python.org/pypi/autopep8 ease your task. 

Thanks for the contribution.

[1] https://travis-ci.org/pythonindia/junction/jobs/125199986#L957

File: tests/unit/tickets/management/commands/test_sync_data.py (hunk 1 above)
Comment (raun) [inline tests/unit/tickets/management/commands/test_sync_data.py]: Let me change it and resubmit the patch. Thanks for pointing that out.

**Comment by theskumar** [comment]
Closed by #431

**Pull request closed without merging**